partner-chains-cli/src/prepare_configuration/prepare_main_chain_config.rs

```diff
@@ -74,11 +74,11 @@ fn run_pc_contracts_cli_addresses<C: IOContext>(
 		&kupo_and_ogmios_config,
 		cardano_network,
 	);
-	let addresses_string = context.run_command(&cmd)?;
+	let addresses_string = context
+		.run_command(&cmd)
+		.context("Failed to fetch data from Ogmios or Kupo. Please check connection configuration and try again.")?;
```

Comment:
Isn't it false assumption? Usually, it will be the cause, but will it always be? What if payment key points to vkey instead of skey?

Reply:
You're right. I wanted to keep to the original errors at first. It might be best to just remove this, as the actual error from smart-contracts-cli will be logged too.